packages/worker-api/src/integriteeWorker.spec.ts

```diff
@@ -78,11 +79,25 @@ describe('worker', () => {
 
         describe('accountInfoGetter', () => {
             it('should return value', async () => {
-                const getter = await worker.accountInfoGetter(charlie, network.shard);
+                const getter = await worker.accountInfoGetter(alice, network.shard);
                 console.log(`AccountInfoGetter: ${JSON.stringify(getter)}`);
                 const result = await getter.send();
                 console.log('getAccountInfo:', result.toHuman());
                 expect(result).toBeDefined();
+                const info = result as AccountInfo;
+                expect(info.data.free.toBigInt()).toBeGreaterThan(0);
+            });
+
+            it('should fall back to default if signed by unauthorized delegate', async () => {
+                const getter = await worker.accountInfoGetter(alice, network.shard, { delegate: charlie });
+                console.log(`AccountInfoGetter with unauthorized signature: ${JSON.stringify(getter)}`);
+                const result = await getter.send();
+                console.log('getAccountInfo:', result.toHuman());
+                expect(result).toBeDefined();
+                const info = result as AccountInfo;
+                console.log("parsed: ", info.data.free);
+                // we don't forward errors here. instead, failures are mapped to default, which is zero
+                expect(info.data.free.toBigInt()).toEqual(BigInt(0));
```

Comment:
toBeDefined is used too often IMHO. It says almost nothing about success

Reply:
Yes, but I did this as I wanted to test it against a running setup that might return different results for sequential runs.